tools/pd-simulator/simulator/cases/redundant_balance_region.go

```diff
@@ -0,0 +1,89 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// //     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cases
+
+import (
+	"github.com/pingcap/kvproto/pkg/metapb"
+	"github.com/pingcap/pd/server/core"
+	"github.com/pingcap/pd/tools/pd-simulator/simulator/dto"
+	"github.com/pingcap/pd/tools/pd-simulator/simulator/simutil"
+	"time"
+)
+
+func newRedundantBalanceRegion() *Case {
+	var simCase Case
+
+	storeNum := simutil.CaseConfigure.StoreNum
+	regionNum := simutil.CaseConfigure.RegionNum * storeNum / 3
+	if storeNum == 0 || regionNum == 0 {
+		storeNum, regionNum = 6, 4000
```

Comment:
Can we only use greater or equal 4000 regions to replay the redundant balance case?

Reply:
Less is also meaningful. When using adjustTolerantRatio and region num is less than 4000, there also will happen redundant balance. It does not look normal until setting tolerantRatio with 50.